packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -14,6 +17,14 @@ contract L1Block is Semver {
     /// @notice Address of the special depositor account.
     address public constant DEPOSITOR_ACCOUNT = 0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001;
 
+    /// @notice Storage slot with the address of the current block hashes offset.
+    /// @dev This is the keccak-256 hash of "l1block.block_hashes_storage_offset".
+    bytes32 private constant BLOCK_HASHES_STORAGE_OFFSET =
+        0xdcb9291dfcbfcfdf70cc580ec1c1835768ebd0ed4a668ccb9e9e6167f533e655;
+
+    /// @notice The max count of block hashes to store.
+    uint16 public constant BLOCK_HASHES_SIZE = 50000;
```

Comment:
This should be a power of 2

Reply:
We have benchmarked the two implementations and it turns out that they have the exact same gas profile.  I am genuinely curious to understand why you prefer to use the implementation that you have referenced instead of the current one?